### DISCUSSION TITLE
Maximum update depth exceeded. Involves `createNotifyManager` in `notifyManager.js`

### DISCUSSION THREAD

**Discussion by kavitang** [TanStack/query#7089] [Q&A]
Since we upgraded to react query v5 our app has been throwing a Maximum depth exceeded error continually.  This does not seem to be impacting our app, but it is a cause for concern. I'm unable to replicate this in a sandbox environment, and something in our app is triggering this. Please find the error below

```
Uncaught Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.
    at throwIfInfiniteUpdateLoopDetected (react-dom.development.js:26657:11)
    at getRootForUpdatedFiber (react-dom.development.js:7679:3)
    at enqueueConcurrentRenderForLane (react-dom.development.js:7601:10)
    at forceStoreRerender (react-dom.development.js:11914:14)
    at handleStoreChange (react-dom.development.js:11893:7)
    at eval (notifyManager.js:40:9)
    at notifyFn (notifyManager.js:6:5)
    at eval (notifyManager.js:51:13)
    at Array.forEach (<anonymous>)
    at eval (notifyManager.js:50:25)
    at batchNotifyFn (notifyManager.js:9:5)
    at eval (notifyManager.js:49:9)
```
<img width="1478" alt="Screenshot 2024-03-12 at 7 50 43 AM" src="https://github.com/TanStack/query/assets/96101675/961de784-0ed7-43e6-874c-5ac8c9fe201f">

It seems to be triggered from `useBaseQuery`

```
React.useSyncExternalStore(
    React.useCallback(
      (onStoreChange) => {
        const unsubscribe = isRestoring ? () => void 0 : observer.subscribe(notifyManager.batchCalls(onStoreChange)); <----
        observer.updateResult();
        return unsubscribe;
      },
      [observer, isRestoring]
    ),
    () => observer.getCurrentResult(),
    () => observer.getCurrentResult()
  );
```
This is in a Next js app (v14), running react-query version `5.26.3`.

I would really appreciate any hints as to what could be causing this or any leads into potential sources. Our application is large and we use hooks across the app liberally. This issue was not present in v4.

Thanks

**Comment by maiconcarraro** [comment]
Got a similar sentry report:
![image](https://github.com/user-attachments/assets/6dbb6df2-8f9f-4bf1-8f70-9c4b59c1db21)
